shadows/framework/src/main/java/org/robolectric/shadows/ShadowBitmapFactory.java

```diff
@@ -231,6 +232,11 @@ public static Bitmap create(final String name, final BitmapFactory.Options optio
       options.outWidth = p.x;
       options.outHeight = p.y;
     }
+
+    if (shadowBitmap.getDensity() <= Bitmap.DENSITY_NONE) {
```

Comment:
Does this mean that all bitmaps created with BitmapFactory.decodeByteArray will have their density set to DisplayMetrics.DENSITY_DEVICE? Aren't there valid use cases for DENSITY_NONE?

Reply:
Should this instead check for the density-related fields in `BitmapFactory.Options opts`?

Reply:
Thank you for your reply. Your suggestion is right. I rewrote the code and added test. Please review again.